### ISSUE TITLE
Mouse is not being captured

### ISSUE THREAD

**Issue by rexy712** [minetest/minetest#8539]
##### Issue type
 Bug report
##### Minetest version
```
5.0.1
```

##### OS / Hardware

Operating system: Gentoo GNU/Linux
CPU: Ryzen 1800X

GPU model: Vega 64
OpenGL version: 4.6

##### Summary

Whenever my mouse cursor enters the minetest window during gameplay, the cursor will jump to the center of my primary monitor. If I put the minetest window at the center of my primary monitor so that the cursor lands inside of it, the player can only face the sky and spin very quickly.

The issue does not start until a world is loaded and gameplay begins. If I manage to open the pause menu, the issue stops immediately. After unpausing, the issue starts again.

Enabling fullscreen does not solve the issue. The mouse is unable to leave the minetest window, but I can still only face the sky and spin.

I think it may be relevant that I'm running i3-gaps as my window manager.

##### Steps to reproduce

1) Build the game from source as described in the readme or install via gentoo package manager.
2) Run the game.
3) Create a new world and 'Play Game'
4) Once the world is loaded, the problem starts.

**Comment by paramat** [comment]
It is worth searching through MT issues for similar reports, i seem to remember it is often caused by the user's setup, not MT.

**Comment by rexy712** [comment]
I searched through the issues for anything mentioning "mouse". I found one similar that said fullcreen fixed it for them. Fullscreening didn't work for me though.
I'll try it out on another window manager to see if some of my i3 settings are the cause of the problem.

**Comment by rexy712** [comment]
I decided to check my xorg.conf files and remembered I had my mouse's transformation matrix changed from default:
```
Section "InputClass"
        Identifier "mouse"
        MatchIsPointer "yes"
        Option "AccelerationProfile" "-1"
        Option "AccelerationScheme" "none"
        Option "AccelSpeed" "-1"
        Option "TransformationMatrix" "0.5 0 0 0 0.5 0 0 0 1"
EndSection
```

Changing the acceleration options doesn't change the issue, but setting the transformation matrix back to the default of '1 0 0 0 1 0 0 0 1' using xinput makes the mouse work properly and the game is playable.

**Comment by sfan5** [comment]
This is most likely an upstream issue since Irrlicht is responsible for the window/input handling.

**Comment by triffid** [comment]
Is there an upstream bug we can track, or should we create one?

Needing to set my trackball so it can cross my thumb more than twice to get the cursor across the screen is a pretty abysmal penalty for being able to play minetest.

**Comment by sfan5** [comment]
Development of Irrlicht is extremely slow, but you can try your luck on [their bugtracker](https://sourceforge.net/p/irrlicht/bugs/).
Though before doing this it would be good if someone takes the time to confirm that it's not Minetest's fault.

**Comment by Mathiaszero** [comment]
There's a feature I use on virtualbox called mouse integration; based on the description of original post, I too have a mouse movement issue where slight adjustments of my mouse cause the character rotate all over the place, often staring at the sky or facing downwards. I wonder if it would be helpful to check how the disabling of mouse integration works, and possibly implement that feature to the engine as an option?

**Comment by vikaig** [comment]
I have the same problem, I confirm that the problem is due to TransformationMatrix, created a report in the upstream bug tracker: https://sourceforge.net/p/irrlicht/bugs/450/

**Comment by randomMesh** [comment]
I don't think this is an Irrlicht bug, i have the same problem with a plain OpenGL app under FreeBSD, using GLFW as the context creation framework.

The problem must lie deeper (libinput or something).

**Comment by sirnewton01** [comment]
I have what appears to be a similar problem compiling from the dev branch of minetest and irrlicht. This seems to be a regressions since the homebrew version of minetest 5.4.x seems fine in this respect. There's another bug related to the rendering though, which makes it very hard to play.
